exporter/SynthesisFusionAddin/src/UI/GeneralConfigTab.py

```diff
@@ -0,0 +1,334 @@
+            weightInput.isVisible = weightTableInput.isVisible = False
+            frictionOverrideButton.isVisible = frictionCoefficient.isVisible = False
+
+    @property
+    def exportMode(self) -> ExportMode:
+        exportModeDropdown: adsk.core.DropDownCommandInput = self.generalOptionsTab.children.itemById(
+            "exportModeDropdown"
+        )
+        if exportModeDropdown.selectedItem.index == 0:
+            return ExportMode.ROBOT
+        else:
+            assert exportModeDropdown.selectedItem.index == 1
+            return ExportMode.FIELD
+
+    @property
+    def compress(self) -> bool:
+        compressButton: adsk.core.BoolValueCommandInput = self.generalOptionsTab.children.itemById(
+            "compressOutputButton"
+        )
+        return compressButton.value
+
+    @property
+    def exportAsPart(self) -> bool:
+        exportAsPartButton: adsk.core.BoolValueCommandInput = self.generalOptionsTab.children.itemById(
+            "exportAsPartButton"
+        )
+        return exportAsPartButton.value
+
+    @property
+    def selectedUnits(self) -> PreferredUnits:
+        return self.currentUnits
+
+    @property
+    def robotWeight(self) -> KG:
+        weightInput: adsk.core.ValueCommandInput = self.generalOptionsTab.children.itemById(
+            "weightTable"
+        ).getInputAtPosition(0, 1)
+        if self.currentUnits == PreferredUnits.METRIC:
+            return KG(weightInput.value)
+        else:
+            assert self.currentUnits == PreferredUnits.IMPERIAL
+            return toKg(weightInput.value)
+
+    @property
+    def autoCalculateWeight(self) -> bool:
+        autoCalcWeightButton: adsk.core.BoolValueCommandInput = self.generalOptionsTab.children.itemById(
+            "autoCalcWeightButton"
+        )
+        return autoCalcWeightButton.value
+
+    @property
+    def exportLocation(self) -> ExportLocation:
+        exportLocationDropdown: adsk.core.DropDownCommandInput = self.generalOptionsTab.children.itemById(
+            "exportLocation"
+        )
+        if exportLocationDropdown.selectedItem.index == 0:
+            return ExportLocation.UPLOAD
+        else:
+            assert exportLocationDropdown.selectedItem.index == 1
+            return ExportLocation.DOWNLOAD
+
+    @property
+    def overrideFriction(self) -> bool:
+        overrideFrictionButton: adsk.core.BoolValueCommandInput = self.generalOptionsTab.children.itemById(
+            "frictionOverride"
+        )
+        return overrideFrictionButton.value
+
+    @property
+    def frictionOverrideCoeff(self) -> float:
+        frictionSlider: adsk.core.FloatSliderCommandInput = self.generalOptionsTab.children.itemById(
+            "frictionCoefficient"
+        )
+        return frictionSlider.valueOne
+
+    @logFailure
+    def handleInputChanged(self, args: adsk.core.InputChangedEventArgs) -> None:
+        commandInput = args.input
+        if commandInput.id == "exportModeDropdown":
```

Comment:
Since you're matching to exact values, this could be a switch statement. If the performance cost doesn't matter, disregard this comment.

Reply:
Reason for me avoiding that here was that the line lengths would be an additional 4 characters longer, and this function already has some really long lines in it. But yes, you are absolutely correct.